Amqp/src/Akka.Streams.Amqp/Dsl/AmqpSource.cs

```diff
@@ -5,14 +5,32 @@ namespace Akka.Streams.Amqp.Dsl
     public static class AmqpSource
     {
         /// <summary>
-        /// Creates an <see cref="AmqpSource"/> with given settings and buffer size.
+        /// The `committableSource` makes it possible to commit (ack/nack) messages to RabbitMQ.
+        /// This is useful when "at-least once delivery" is desired, as each message will likely be
+        /// delivered one time but in failure cases could be duplicated.
+        /// If you commit the offset before processing the message you get "at-most once delivery" semantics,
+        /// and for that there is a [[#atMostOnceSource]].
+        /// Compared to auto-commit, this gives exact control over when a message is considered consumed.
         /// </summary>
         /// <param name="settings"></param>
         /// <param name="bufferSize"></param>
         /// <returns>TBD</returns>
-        public static Source<IncomingMessage, NotUsed> Create(IAmqpSourceSettings settings, int bufferSize)
+        public static Source<CommittableIncomingMessage, NotUsed> CommittableSource(IAmqpSourceSettings settings, int bufferSize)
         {
             return Source.FromGraph(new AmqpSourceStage(settings, bufferSize));
-        } 
+        }
+
+        /// <summary>
+        /// Convenience for "at-most once delivery" semantics. Each message is acked to RabbitMQ
+        /// before it is emitted downstream.
+        /// </summary>
+        /// <param name="settings"></param>
+        /// <param name="bufferSize"></param>
+        /// <returns></returns>
+        public static Source<IncomingMessage, NotUsed> AtMostOnceSource(IAmqpSourceSettings settings, int bufferSize)
+        {
+            return CommittableSource(settings, bufferSize)
+                .SelectAsync(1, cm => cm.Ack().Task.ContinueWith(_ => cm.Message));
```

Comment:
Maybe don't risk and use async lambda? What if `Ack().Task` will throw or timeout? Exception would be swallowed.

Reply:
Done. However, I'm not sure I understand the difference :)